x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -303,4 +303,66 @@ mod tests {
                 .next()
         );
     }
+
+    #[test]
+    fn wrong_assertion() {
```

Comment:
```suggestion
    fn assertion_can_fail() {
```
We shouldn't be testing this here though. There should be tests on the Go side for this.

Reply:
Would you remove this ? This is a property that I broke during the writing of this feature and would have been great to figure out what is missing more quickly with an already written test. If it's written on the Go side, I guess that it would be interacting more directly with the simulator and anything that is between Go and Rust (so essentially communication) is left and could potentially be buggy.

Reply:
renamed the test

Reply:
The problem is that the test is in the counter example, but you're not testing the counter example, you're testing the simulator itself. You're coupling two crates together instead of testing at the boundaries.